### PR TITLE
feat: add LIST command

### PR DESCRIPTION
**Pull request by theovgl**

### CODE REVIEW

File: src/IRCReplies.cpp [hunk 1]
```diff
@@ -100,6 +100,53 @@ RPL_WHOISCHANNELS::RPL_WHOISCHANNELS(Peer &peer, const Peer &target, const std::
 	this->input = this->updateInputFromFields();
 }
 
+RPL_LIST::RPL_LIST(Peer &peer, const std::string &channel, const std::string &topic, bool include_prefix) : Message(peer, std::string())
+{
+	this->command = "322";
+	this->arguments[0] = peer.getNickname();
+	this->arguments[1] = channel;
+	this->arguments[2] = ":" + topic;
```
Comment (ppiques) [inline src/IRCReplies.cpp:108]: I'm not sure the ":" is necessary here

File: src/commands/list.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include "commands.hpp"
+
+static void	list_all_channels(Message &message, Dependencies &deps)
+{
+	for (std::map<std::string, Channel>::iterator chan_it = deps.channels.begin(); chan_it != deps.channels.end(); ++chan_it)
+	{
+		message.peer.sendMessage(RPL_LIST(message.peer, chan_it->second.stringify(), chan_it->second.getTopic()));
+	}
```
Comment (ppiques) [inline src/commands/list.cpp:8]: are the brackets necessary ?

**Review by ppiques — COMMENTED** [review]
small refactors suggestions

File: src/commands/list.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#include "commands.hpp"
+
+static void	list_all_channels(Message &message, Dependencies &deps)
+{
+	for (std::map<std::string, Channel>::iterator chan_it = deps.channels.begin(); chan_it != deps.channels.end(); ++chan_it)
```
Comment (vfurmane) [inline src/commands/list.cpp:5]:
```suggestion
	for (ChannelManager::iterator chan_it = deps.channels.begin(); chan_it != deps.channels.end(); ++chan_it)
```

File: src/commands/list.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+			if (_base_channel::isValidName(*channel_it))
+			{
+				_base_channel base_channel = Channel::parse(*channel_it);
+				if (deps.channels.has(base_channel.getName()))
+				{
+					Channel &channel = deps.channels[base_channel.getName()];
+					if (channel.getTopic().empty())
+						message.peer.sendMessage(RPL_LIST(message.peer, channel.stringify(), ""));
+					else
+						message.peer.sendMessage(RPL_LIST(message.peer, channel.stringify(), channel.getTopic()));
```
Comment (vfurmane) [inline src/commands/list.cpp:31]: The condition is useless, if the topic is empty, topic is then equal to `""`

File: test/commands/list.test.cpp [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#include "catch.hpp"
+
+#include "../src/commands/list.cpp"
```
Comment (vfurmane) [inline test/commands/list.test.cpp:3]: Nice tests you lazy jackass

**Review by vfurmane — CHANGES_REQUESTED** [review]

**Review by ppiques — APPROVED** [review]

**Review by vfurmane — APPROVED** [review]

**Pull request merged**